lib/pdn/pdn.c

```diff
@@ -692,6 +698,17 @@ static void on_cfun(int mode, void *ctx)
 	}
 
 	if (mode == MODEM_CFUN_POWER_OFF) {
+		k_mutex_lock(&list_mutex, K_FOREVER);
+		SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&pdn_contexts, pdn, tmp, node) {
+			if (pdn->callback) {
+				pdn->callback(pdn->context_id, PDN_EVENT_CTX_DESTROYED, 0);
+			}
+
+			sys_slist_find_and_remove(&pdn_contexts, &pdn->node);
+			k_free(pdn);
+		}
+		k_mutex_unlock(&list_mutex);
+
 #if defined(CONFIG_PDN_DEFAULTS_OVERRIDE)
 		pdn_defaults_override();
 #endif
```

Comment:
I think these should be overridden only when changing to `MODEM_CFUN_NORMAL`, `MODEM_CFUN_ACTIVATE_LTE`.

Reply:
PDP configuration is lost on modem side on cfun=0. Therefore I think it makes sense to do it on CFUN=0.

Reply:
Alright time for another coffee ☕